src/xopen/__init__.py

```diff
@@ -85,23 +86,18 @@ def __del__(self):
             pass
 
 
-class PipedGzipWriter(Closing):
+class PipedCompressionWriter(Closing):
     """
-    Write gzip-compressed files by running an external gzip or pigz process and
-    piping into it. pigz is tried first. It is fast because it can compress using
-    multiple cores.
-
-    If pigz is not available, a gzip subprocess is used. On Python 2, this saves
-    CPU time because gzip.GzipFile is slower. On Python 3, gzip.GzipFile is on
-    par with gzip itself, but running an external gzip can still reduce wall-clock
-    time because the compression happens in a separate process.
+    Write Compressed files by running an external process and piping into it.
     """
-
-    def __init__(self, path, mode='wt', compresslevel=6, threads=None):
+    def __init__(self, path, program, mode='wt',
+                 compresslevel: Optional[int] = None,
+                 threads_flag: str = None,
+                 threads: Optional[int] = None):
         """
         mode -- one of 'w', 'wt', 'wb', 'a', 'at', 'ab'
-        compresslevel -- gzip compression level
-        threads (int) -- number of pigz threads. If this is set to None, a reasonable default is
+        compresslevel -- compression level
+        threads (int) -- number of threads. If this is set to None, a reasonable default is
             used. At the moment, this means that the number of available CPU cores is used, capped
             at four to avoid creating too many threads. Use 0 to let pigz use all available cores.
```

Comment:
This still refers to pigz. Also, the `threads_flag` parameter is not documented.

Reply:
I removed the pigz references, moved the pigz specific stuff to the docstring for PipedGzipWriter. I also made the entire `PipedCompressionWriter` class private (`_PipedCompressionWriter`) and did the same for PipedCompressionReader.